src/System.Text.Json/src/System/Text/Json/Writer/JsonWriterHelper.Date.cs

```diff
@@ -89,11 +90,18 @@ public static void TrimDateTimeOffset(Span<byte> buffer, out int bytesWritten)
                     // Last index of the offset
                     int bufferEnd = curIndex + 5;
 
-                    buffer[bufferEnd] = buffer[32];
-                    buffer[bufferEnd - 1] = buffer[31];
-                    buffer[bufferEnd - 2] = buffer[30];
-                    buffer[bufferEnd - 3] = buffer[29];
-                    buffer[bufferEnd - 4] = buffer[28];
+                    // Temporary offset characters to prevent their values from being overwritten
```

Comment:
Is the only place we need to store in locals to avoid overwriting the original input? If so, can we get away with only saving "half" of them (say just the minute digits)?

Reply:
Yes this is the only place. We don't touch anything before the period, and we cache the fraction digits.

When we write 2-6 fraction digits, the hour digits and first minute digit could be overwritten. We don't need to cache the second minute digit.